### PR TITLE
ref(open-pr-comments): use parser classes instead of standalone regexes

### PR DESCRIPTION
**Pull request by cathteng**
Since we are planning on expanding Open PR Comments to more languages, we should have a mapping from supported file extensions to the appropriate parser to extract function names from the file patch (diff), rather than the single regex and extension check we do currently for Python.

### CODE REVIEW

File: src/sentry/tasks/integrations/github/open_pr_comment.py [hunk 1]
```diff
@@ -189,7 +189,7 @@ def safe_for_comment(
     for file in pr_files:
         filename = file["filename"]
         # don't count the file if it was added or is not a Python file
```
Comment (cathteng) [inline src/sentry/tasks/integrations/github/open_pr_comment.py:191]:
```suggestion
        # don't count the file if it was added or is not a supported language
```

**Review by nhsiehgit — COMMENTED** [review]
nice this looks good 👍

**Review by nhsiehgit — APPROVED** [review]

File: src/sentry/tasks/integrations/github/patch_parsers.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import re
+from abc import abstractmethod
+from typing import Set
+
+
+class LanguageParser:
```
Comment (schew2381) [inline src/sentry/tasks/integrations/github/patch_parsers.py:6]: Does this need to be declared as an abstract class?

Comment (cathteng) [reply src/sentry/tasks/integrations/github/patch_parsers.py:6]: i don't think there's a decorator for an abstract class? each method should have the `@abstractmethod` decorator

Comment (schew2381) [reply src/sentry/tasks/integrations/github/patch_parsers.py:6]:
```
from abc import ABC

class LanguageParser(ABC):
```
I think it'd be like this

Comment (cathteng) [reply src/sentry/tasks/integrations/github/patch_parsers.py:6]: oh i see. yes i can add this

**Pull request merged**